src/status_im/ui/screens/wallet/send/views.cljs

```diff
@@ -45,80 +43,92 @@
           :secure-text-entry      true
           :placeholder            (i18n/label :t/enter-password)
           :placeholder-text-color "#939ba1"
           :on-change-text         #(do
                                      (re-frame/dispatch [:set-in [:wallet/send-transaction :password] %]))
-          :style                  styles/password}]]]
+          :style                  send.styles/password}]]]
      (when wrong-password?
        [components/tooltip (i18n/label :t/wrong-password)])]))
 
-
 (defview signing-buttons []
   (letsubs [sign-enabled? [:wallet.send/sign-password-enabled?]]
     [react/view wallet.styles/buttons-container
      [react/touchable-highlight {:on-press #(re-frame/dispatch [:wallet/discard-transaction])}
       [react/view (wallet.styles/button-container true)
        [components/button-text (i18n/label :t/cancel)]]]
      [react/view components.styles/flex]
+
      [react/touchable-highlight {:on-press #(re-frame/dispatch [:wallet/sign-transaction])}
       [react/view (wallet.styles/button-container sign-enabled?)
        [components/button-text (i18n/label :t/transactions-sign-transaction)]
-       [vi/icon :icons/forward {:color :white :container-style wallet.styles/forward-icon-container}]]]]))
+       [vector-icons/icon :icons/forward {:color :white :container-style wallet.styles/forward-icon-container}]]]]))
 
-(defview sign-buttons []
-  (letsubs [sign-enabled? [:wallet.send/sign-enabled?]]
+(defn sign-enabled? [amount-error to-address amount]
+  (and
+    (nil? amount-error)
+    (not (nil? to-address)) (not= to-address "")
+    (not (nil? amount)) (not= amount "")))
+
+(defn- sign-buttons [amount-error to-address amount sufficient-funds?]
+  (let [sign-enabled? (sign-enabled? amount-error to-address amount)
+        immediate-sign-enabled? (and sign-enabled? sufficient-funds?)]
     [react/view wallet.styles/buttons-container
      (when sign-enabled?
        [react/touchable-highlight {:on-press sign-later}
         [react/view (wallet.styles/button-container sign-enabled?)
          [components/button-text (i18n/label :t/transactions-sign-later)]]])
      [react/view components.styles/flex]
-     [react/touchable-highlight {:on-press (when sign-enabled? #(re-frame/dispatch [:set-in [:wallet/send-transaction :signing?] true]))}
-      [react/view (wallet.styles/button-container sign-enabled?)
+     [react/touchable-highlight {:on-press (when immediate-sign-enabled? #(re-frame/dispatch [:set-in [:wallet/send-transaction :signing?] true]))}
+      [react/view (wallet.styles/button-container immediate-sign-enabled?)
        [components/button-text (i18n/label :t/transactions-sign-transaction)]
-       [vi/icon :icons/forward {:color :white :container-style wallet.styles/forward-icon-container}]]]]))
+       [vector-icons/icon :icons/forward {:color :white :container-style wallet.styles/forward-icon-container}]]]]))
+
+(defn- sufficient-funds? [amount balance]
+  (<= amount (money/wei->ether balance)))
```

Comment:
what about transaction fee?

Reply:
Make sense.

Reply:
I will handle this in a separate ticket.